client/my-sites/upgrades/domain-management/transfer/transfer-to-other-site/index.jsx

```diff
@@ -0,0 +1,167 @@
+	static propTypes = {
+		selectedDomainName: React.PropTypes.string.isRequired,
+		selectedSite: React.PropTypes.object.isRequired,
+		currentUser: React.PropTypes.object.isRequired
+	};
+
+	state = {
+		targetSiteSlug: '',
+		showConfirmationDialog: false,
+		disableDialogButtons: false
+	};
+
+	isDataReady() {
+		return this.props.domains.hasLoadedFromServer;
+	}
+
+	isSiteEligible = ( site ) => {
+		return site.capabilities.manage_options &&
+			! site.jetpack &&
+			! get( site, 'options.is_domain_only', false ) &&
+			! ( this.props.domainsWithPlansOnly && get( site, 'plan.product_slug' ) === PLAN_FREE ) &&
+			site.ID !== this.props.selectedSite.ID;
+	}
+
+	handleSiteSelect = ( targetSiteSlug ) => {
+		this.setState( {
+			targetSiteSlug,
+			showConfirmationDialog: true,
+		} );
+	}
+
+	handleConfirmTransfer = ( targetSite, closeDialog ) => {
+		const { selectedDomainName } = this.props;
+		const targetSiteName = targetSite.name;
+		const successMessage = this.props.translate(
+				'%(selectedDomainName)s has been transferred to site: %(targetSiteName)s',
+				{ args: { selectedDomainName, targetSiteName } } );
+		const defaultErrorMessage = this.props.translate(
+				'Failed to transfer %(selectedDomainName)s, please try again or contact support.', {
+					args: { selectedDomainName } } );
+
+		this.setState( { disableDialogButtons: true } );
+		wpcom.transferToSite( this.props.selectedSite.ID, this.props.selectedDomainName, targetSite.ID )
+			.then(
+				() => {
+					this.props.successNotice( successMessage, { duration: 10000, isPersistent: true } );
+					page( paths.domainManagementList( this.props.selectedSite.slug ) );
+				}, ( error ) => {
+					this.setState( { disableDialogButtons: false } );
+					closeDialog();
+					this.props.errorNotice( error.message || defaultErrorMessage );
+				}
+			);
+	}
+
+	handleDialogClose = () => {
+		if ( ! this.state.disableDialogButtons ) {
+			this.setState( { showConfirmationDialog: false } );
+		}
+	}
+
+	render() {
+		if ( ! this.isDataReady() ) {
+			return <DomainMainPlaceholder goBack={ this.goToEdit } />;
+		}
+
+		const { selectedSite, selectedDomainName } = this.props;
+		const { slug } = selectedSite;
+
+		return (
+			<Main className="transfer-to-other-site">
+				<Header
+					selectedDomainName={ selectedDomainName }
+					backHref={ paths.domainManagementTransfer( slug, selectedDomainName ) }>
+					{ this.props.translate( 'Transfer Domain To Another Site' ) }
+				</Header>
+				{ this.renderSection() }
+			</Main>
+		);
+	}
+
+	renderSection() {
+		const { currentUserCanManage } = getSelectedDomain( this.props );
+		if ( ! currentUserCanManage ) {
+			return <NonOwnerCard { ...omit( this.props, [ 'children' ] ) } />;
+		}
+
+		const { selectedDomainName: domainName, domainsWithPlansOnly, translate } = this.props;
+		let message;
+		if ( domainsWithPlansOnly ) {
+			message = translate( 'Please choose a site with a paid plan ' +
```

Comment:
Added the `site with a paid plan` part when we're dealing with a DWPO user to hint why some sites are hidden.